tests/rptest/tests/delete_records_test.py

```diff
@@ -139,10 +139,19 @@ def _apply_local_retention_settings(self):
             self.local_retention)
 
     def get_topic_info(self, topic_name):
-        topics_info = list(self.rpk.describe_topic(topic_name))
-        self.logger.info(topics_info)
-        assert len(topics_info) == 1
-        return topics_info[0]
+        def describe_topic():
+            topics_info = list(self.rpk.describe_topic(topic_name))
+            if len(topics_info) > 0:
+                self.logger.info(topics_info)
+                return True, topics_info[0]
+            else:
+                return False
```

Comment:
I'm thinking maybe it would be a more impactful/useful to have the wait be in _create_initial_topics. WDYT?

Reply:
I think the wait needs to be here. From what I can tell, the issue here is not that there is a race between the topic creation and the `describe_topics()` call.

 [`rpk.describe_topic()`](https://github.com/redpanda-data/redpanda/blob/599f24f2fb755f8ac5d82ae76fc225ff8548829f/tests/rptest/clients/rpk.py#L730-L731) only yields a value if the response it gets back from a redpanda broker is complete.

There are two threads running in this test, one which is moving partitions around. Because of that, there may be a race between `rpk.describe_topic()` and the partition movement which results in `rpk` output like

```
PARTITION  LEADER  EPOCH  REPLICAS  LOG-START-OFFSET          HIGH-WATERMARK
0          1       2      [1 2 3]   NOT_LEADER_FOR_PARTITION  18618
```

In this case, `describe_topic()` won't yield a value, and the assert will fail.

Reply:
Ah thanks, got it. Leaving it as is makes sense